### DISCUSSION TITLE
Button Link

### DISCUSSION THREAD

**Discussion by edssv** [adobe/react-spectrum#3778] [Q&A] [answered]
`<Button elementType={Link}></Button>`
I want to set button type as Link from react router dom, how can I specify props "to"?  It usually looks like this: `<Link to="/"/>`

**Comment by reidbarber** [comment, marked as answer]
The Button component does not support this. We've had this issue come up internally before and here is the general takeaway:

- In general, links should look like links instead of buttons.

- There is a workaround where you can navigate within the Button's onPress handler:

React Router 6.x.x:

```jsx
const navigate = useNavigate();
...
<Button onPress={() => navigate("/newpage")}>
   Click here
</Button>
```

React Router 5.x.x:

```jsx
const history = useHistory();
...
<Button onPress={() => history.push("/newpage")}>
   Click here
</Button>
```

- This isn't ideal, because you lose link semantics, however you are already representing it as a button to visual users anyway.
